src/lib/components/setlist/AddParticipantToSongModal.svelte

```diff
@@ -0,0 +1,85 @@
+<script lang="ts">
+	import { invalidateAll } from '$app/navigation'
+	import CustomModal from '$lib/components/setlist/CustomModal.svelte'
+	import { db } from '$lib/firebase/client/firebase'
+	import type { Song } from '$lib/types/domain/song'
+	import type { user } from '$lib/types/domain/user'
+	import { Form, ComboBox, TextInput } from 'carbon-components-svelte'
+	import type { DropdownItem } from 'carbon-components-svelte/types/Dropdown/Dropdown.svelte'
+	import { addDoc, collection, doc } from 'firebase/firestore'
+
+	export let open = false
+	export let onClose: () => void
+
+	export let song: Song
+	export let users: user[]
+
+	let userId: string
+	let instrument: string
+
+	let participantListItems: DropdownItem[] = users.map(
+		(user) => ({ id: user.id, text: user.name } as DropdownItem)
+	)
+
+	async function addParticipantToSong() {
+		try {
+			// Add the song to the songs a user plays on
+			await addDoc(collection(db, 'users/' + userId + '/playsSongs'), {
+				song: doc(db, 'songs', song.id),
+				part: instrument
+			})
+
+			// Reset state
+			userId = ''
+			instrument = ''
+
+			invalidateAll()
+			return true
+		} catch (error) {
+			console.error(error) //TODO: handle error
+			return false
+		}
```

Comment:
Dit hele return true als het succes is en anders false verhaal vinnik nogal niet mooi tbh, error moet gewoon hoger gecatched worden denk ik, maar dat kan in een andere issue.

Reply:
Mjah vond het ook niet ideaal, maar vond het idd beetje buiten de scope van deze issue nu